tekton/ci/shared/go_coverage_task.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: go-coverage
+  annotations:
+    description: |
+      Clones a repository from github into an empty-dir workspace.
+spec:
+  params:
+    - name: prNumber
+      description: The number of the pull request to check out.
+    - name: branchName
+      description: Branch of the pull request to check out.
+    - name: pullSha
+      description: sha of the commit making the pull request
+    - name: baseSha
+      description: sha of the commit making the pull request against.
+    - name: jobType
+      description: Type of prow job - presubmit/postsubmit/periodic.
+    - name: buildNumber
+      description: Prow type build ID.
+    - name: repoName
+      description: name of the repository
+    - name: repoOwner
+      description: owner of the repo (user/orgaization).
+    - name: jobName
+      description: name of the job.
+  steps:
+    - name: clone-repo
+      image: alpine/git
+      script: |
+        #!/bin/sh
+        git clone https://github.com/$(params.repoOwner)/$(params.repoName).git
+        cd $(params.repoName)
+        git fetch origin pull/$(params.prNumber)/head:$(params.branchName)
+        git checkout $(params.branchName)
+      workingDir: "$(workspaces.shared-workspace.path)"
+    - name: run-coverage
+      image: gcr.io/tekton-releases/dogfooding/coverage:latest
+      env:
+        - name: PULL_PULL_SHA
+          value: $(params.pullSha)
+        - name: PULL_BASE_SHA
+          value: $(params.baseSha)
+        - name: PULL_NUMBER
+          value: $(params.prNumber)
+        - name: JOB_TYPE
+          value: $(params.jobType)
+        - name: BUILD_NUMBER
+          value: $(params.buildNumber)
+        - name: REPO_NAME
+          value: $(params.repoName)
+        - name: REPO_OWNER
+          value: $(params.repoOwner)
+        - name: JOB_NAME
+          value: $(params.jobName)
+      command:
+        - /coverage
+      args:
+        - "--postsubmit-gcs-bucket=tekton-prow"
+        - "--postsubmit-job-name=post-tekton-pipeline-go-coverage"
+        - "--artifacts=$(workspaces.shared-workspace.path)/artifacts"
+        - "--profile-name=coverage_profile.txt"
+        - "--cov-target=."
+        - "--cov-threshold-percentage=0"
+        - "--github-token=$(workspaces.github-token.path)/token"
+      workingDir: "$(workspaces.shared-workspace.path)/$(params.repoName)"
+  workspaces:
+  - name: shared-workspace
+    mountPath: /go/src/github.com/tektoncd
+  - name: github-token
```

Comment:
Would be useful to provide a description of the workspace and where it expects the token to be provided

Reply:
Yes. And for others too. I just pushed the descriptions.